File: data-structures/lists/c-sharp/nested_lists_update_sub.cs
/*
Raspberry Pi Foundation
Developed as part of Ada Computer Science 
Usage licensed under CC BY-NC-SA 4.0

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;
using System.Collections.Generic;

namespace AdaCodeSamples {
    class NestedLists {
        // Index values of each astronaut
        public const int CraftIndex = 0;
        public const int NameIndex = 1;
        public const int DaysIndex = 2;


        // The Main method is the entry point for all C# programs
        public static void Main() {
            // Declare and initialise the nested list of astronauts
            List<List<string>> allAstronauts = new List<List<string>>() {
                new List<string>() { "ISS", "Sergey Prokopyev", "176" },
                new List<string>() { "ISS", "Dmitry Petelin", "176" },
                new List<string>() { "ISS", "Frank Rubio", "176" },
                new List<string>() { "Tiangong", "Fei Junlong", "112" },
                new List<string>() { "Tiangong", "Deng Qingming", "107" },
                new List<string>() { "Tiangong", "Zhang Lu", "107" },
                new List<string>() { "ISS", "Stephen Bowen", "54" },
                new List<string>() { "ISS", "Warren Hoburg", "14" },
                new List<string>() { "ISS", "Sultan Al Neyadi", "14" },
                new List<string>() { "ISS", "Andrey Fedyaev", "14" }
            };

            // Set the position to be the first astronaut at index 0
            int position = 0;

            // Ask the user for the new number of days in space
            Console.WriteLine("Enter the new number of days:");
            string newDays = Console.ReadLine();

            // Update the number of days in space
            UpdateDaysInSpace(allAstronauts, position, newDays);

            // Output the updated number of days
            Console.WriteLine($"The days has been updated to {allAstronauts[0][2]}");
        }

        
        // Update the number of days in space for an astronaut
        public static void UpdateDaysInSpace(List<List<string>> allAstronauts,
            int astronautIndex, string newDays) {
            
            allAstronauts[astronautIndex][DaysIndex] = newDays;
        }
    }
}

File: data-structures/lists/visual-basic/nested_lists_update_sub.vb
' Raspberry Pi Foundation
' Developed as part of Ada Computer Science
' Usage licensed under CC BY-NC-SA 4.0
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program

Module Program
    ' Index values of each astronaut
    Public Const CraftIndex As Integer = 0
    Public Const NameIndex As Integer = 1
    Public Const DaysIndex As Integer = 2

    ' The Main subroutine is the default entry point for a VB program
    Sub Main()
        ' Declare and initialise the astronaut list
        Dim allAstronauts As New List(Of List(Of String))({
            New List(Of String)({"ISS", "Sergey Prokopyev", "176"}),
            New List(Of String)({"ISS", "Dmitry Petelin", "176"}),
            New List(Of String)({"ISS", "Frank Rubio", "176"}),
            New List(Of String)({"Tiangong", "Fei Junlong", "112"}),
            New List(Of String)({"Tiangong", "Deng Qingming", "107"}),
            New List(Of String)({"Tiangong", "Zhang Lu", "107"}),
            New List(Of String)({"ISS", "Stephen Bowen", "54"}),
            New List(Of String)({"ISS", "Warren Hoburg", "14"}),
            New List(Of String)({"ISS", "Sultan Al Neyadi", "14"}),
            New List(Of String)({"ISS", "Andrey Fedyaev", "14"})
        })

        ' Set the position to be the first astronaut at index 0
        Dim position = 0

        ' Ask the user for the new number of days in space
        Console.WriteLine("Enter the new number of days:")
        Dim newDays As String = Console.ReadLine()

        ' Update the number of days in space
        UpdateDaysInSpace(allAstronauts, position, newDays)

        ' Output the updated number of days
        Console.WriteLine($"The days has been updated to {allAstronauts(0)(2)}")
    End Sub

    ' Update the number of days in space for an astronaut
    Sub UpdateDaysInSpace(ByVal allAstronauts As List(Of List(Of String)),
        ByVal astronautIndex As Integer, ByVal newDays As String)
        
        allAstronauts(astronautIndex)(DaysIndex) = newDays
    End Sub
End Module